source/common/http/header_utility.cc

```diff
@@ -190,5 +190,33 @@ HeaderUtility::requestHeadersValid(const RequestHeaderMap& headers) {
   return absl::nullopt;
 }
 
+bool HeaderUtility::shouldCloseConnection(Http::Protocol protocol,
```

Comment:
At this point we have a few different variants of this function, though the other variants are client side. I'm wondering if it's possible to collapse the other uses into this utility in some way? See

https://github.com/envoyproxy/envoy/blob/f0ebefc2a36746dbf12301d3c258e2f1b66cfe82/source/common/http/http1/conn_pool.cc#L84-L99

and

https://github.com/envoyproxy/envoy/blob/f0ebefc2a36746dbf12301d3c258e2f1b66cfe82/source/common/upstream/health_checker_impl.cc#L342

There might be others also?

Reply:
Good call - those both had variants of the bugs the utility fixes, and thanks to your RequestOrResponseMap it's easy to make the utility consistently usable.